test/e2e/specs/site-editor/browser-history.spec.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+const baseUrl = process.env.WP_BASE_URL || 'http://localhost:8889';
+
+test.describe( 'Site editor browser history', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'Back button works properly', async ( { admin, page } ) => {
+		// Navigate to a template.
+		await admin.visitAdminPage( 'index.php' );
+		await admin.visitSiteEditor();
+
+		expect( page.url() ).toEqual( baseUrl + '/wp-admin/site-editor.php' );
+
+		// Navigate to a single template
+		await page.click( 'role=button[name="Templates"]' );
+		await page.click( 'role=button[name="Index"]' );
+		expect( page.url() ).toEqual(
+			baseUrl +
+				'/wp-admin/site-editor.php?postType=wp_template&postId=emptytheme%2F%2Findex'
+		);
+
+		// Navigate back to the template list
+		await page.goBack();
+		expect( page.url() ).toEqual(
+			baseUrl + '/wp-admin/site-editor.php?path=%2Fwp_template'
+		);
+
+		// Navigate back to the dashboard
+		await page.goBack();
+		await page.goBack();
+		expect( page.url() ).toEqual( baseUrl + '/wp-admin/index.php' );
+	} );
+
+	test( 'Opens the template list from the template details view', async ( {
+		admin,
+		page,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postType: 'wp_template',
+			postId: 'emptytheme//index',
+			canvas: 'edit',
+		} );
+
+		// Navigate to the template list
+		await page.click( 'role=button[name="Show template details"]' );
+		await page.click( 'role=link[name="Manage all templates"]' );
+
+		expect( page.url() ).toEqual(
+			baseUrl + '/wp-admin/site-editor.php?path=%2Fwp_template%2Fall'
+		);
+
+		const title = page.locator( '.edit-site-list-header__title' );
```

Comment:
Using class name selectors is generally not recommended. Try role selectors instead:

```suggestion
		const title = page.getByRole( 'heading', { level: 1 } );
```

Or if you don't need the level 1 heading but want the tab's title:

```js
await expect( page ).toHaveTitle( /^Templates/ );
```

Reply:
Personally, I think that just a random guideline, heading levels or classnames are subject to change similarly. But it's fine, I can change the selector.

Reply:
I would argue that accessibility-wise, heading levels should rarely change (and if it does change, it often represents a change to the user flow too), while class names are implementation details and not exposed to the end users. Let's follow the established [best practices](https://github.com/WordPress/gutenberg/tree/trunk/docs/contributors/code/e2e#use-accessible-selectors) until we change them ❤️ . (Feel free to open an issue/PR to discuss about this!)

Reply:
If that's what you all think we should do, it's fine, it's not important enough for me to lose time on it.